tests/flow/test_udf.py

```
@@ -1807,3 +1807,48 @@ def test_returning_undefined(self):
+        # seconds instead of hanging on the suite's 600s socket timeout
+        bounded = self.env.getConnection()
+        bounded.connection_pool.connection_kwargs['socket_timeout'] = 30
+
+        try:
+            bounded.execute_command("GRAPH.QUERY", GRAPH_ID, "RETURN lib_spin.spin()")
+            raise AssertionError("expected the runaway top-level code to be interrupted")
+        except ResponseError as e:
+            self.env.assertContains("interrupted", str(e).lower())
+        except redis.exceptions.TimeoutError:
+            raise AssertionError(
+                "rebuild ran unbounded: the call neither returned nor errored")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)